generated/nidigital/session.py

```diff
@@ -807,6 +807,43 @@ def configure_voltage_levels(self, vil, vih, vol, voh, vterm):
         errors.handle_error(self, error_code, ignore_warnings=False, is_error_handling=False)
         return
 
+    @ivi_synchronized
+    def get_pin_results_pin_information(self):
+        '''get_pin_results_pin_information
+
+        Returns a list of named tuples (PinInfo) that <FILL IN THE BLANK HERE>
+
+        Fields in PinInfo:
+
+        - **pin_indexes** (int)
+        - **site_numbers** (int)
+        - **channel_indexes** (int)
+
+        Tip:
+        This method requires repeated capabilities (channels). If called directly on the
+        nidigital.Session object, then the method will use all repeated capabilities in the session.
+        You can specify a subset of repeated capabilities using the Python index notation on an
+        nidigital.Session repeated capabilities container, and calling this method on the result.:
+
+            session.channels[0,1].get_pin_results_pin_information()
+
+        Returns:
+            pin_info (list of PinInfo): List of named tuples with fields:
+
+                - **pin_indexes** (int)
+                - **site_numbers** (int)
+                - **channel_indexes** (int)
+
+        '''
+        import collections
+        PinInfo = collections.namedtuple('PinInformation', ['pin_name', 'site_number', 'channel_name'])
+
+        pin_indexes, site_numbers, channel_indexes = self._get_pin_results_pin_information()
+        assert len(pin_indexes) == len(site_numbers), "length of returned arrays don't match"
+        assert len(pin_indexes) == len(channel_indexes), "length of returned arrays don't match"
+
+        return [PinInfo(pin_name=self.get_pin_name(pin_indexes[i]), site_number=site_numbers[i], channel_name=self.get_channel_name(channel_indexes[i])) for i in range(len(pin_indexes))]
```

Comment:
If pins are unmapped, `pin_index` will be returned as -1. Calling `get_pin_name` with -1 throws. So, if `pin_index` == -`1, `pin_name` should be set to empty string.

Reply:
Suggested code change:
```
pin_name= "" if pin_index == -1 else self.get_pin_name(pin_indexes[i])
```

Reply:
Does this apply to channel name too?

Reply:
Fixed